sphinxcontrib/confluencebuilder/translator.py

```diff
@@ -153,10 +155,26 @@ def visit_section(self, node):
     def depart_section(self, node):
         self._section_level -= 1
 
+    def add_secnumber(self, node):
+        # type: (nodes.Element) -> None
+        if node.get('secnumber'):
+            self.body.append('.'.join(map(str, node['secnumber'])) +
+                             self.secnumber_suffix)
+        elif isinstance(node.parent, nodes.section):
+            anchorname = '#' + node.parent['ids'][0]
+            if anchorname not in self.builder.secnumbers:
+                anchorname = ''  # try first heading which has no anchor
+            if self.builder.secnumbers.get(anchorname):
+                numbers = self.builder.secnumbers[anchorname]
+                self.body.append('.'.join(map(str, numbers)) +
+                                 self.secnumber_suffix)
```

Comment:
If we are going to pull code from existing Sphinx implementation, we would need to appropriately add a copyright notice at the top of this file (see also [`confluencebuilder/sphinxcontrib/confluencebuilder/util.py`](https://github.com/sphinx-contrib/confluencebuilder/blob/master/sphinxcontrib/confluencebuilder/util.py#L4) for an example).

Reply:
Good point.

As an alternative, I could directly call `HTML5Translator.add_secnumber()`. That would mean no Sphinx code will need to be copied, but it does tie implementation to the Sphinx implementation (which has upsides and downsides).

Which would you prefer?

Reply:
Copying is fine (with appropriate attribution while also meeting license requirements). Probably better not to rely on other translator's implementation which may cause future issues.